SuperPong/Assets/Scripts/BallController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BallController : MonoBehaviour
+{
+    private Rigidbody sphere;
+    private int velocityRange = 200;
+    private float initialHeight;
+
+	// Use this for initialization
+	void Start ()
+    {
+        sphere = GetComponent<Rigidbody>();
+        initialHeight = sphere.position.y;
+        resetBall(sphere);
+        sphere.AddForce(new Vector3(Random.Range(velocityRange, velocityRange*2), 0, Random.Range(velocityRange, velocityRange*2)));
+    }
+	
+	// Update is called once per frame
+	void Update ()
+    {
+
+    }
+
+    private void OnCollisionExit(Collision collision)
+    {
+        //The qualifier of initial height is needed because the ball sometimes rises a bit
+        //when it bounces off of a wall.
+        if (collision.gameObject.tag.Equals("Floor") && sphere.position.y <= initialHeight)
+        {
+            resetBall(sphere);
+        }
+    }
+
+    private void OnCollisionEnter(Collision collision)
+    {
+        float forceMagnitudeAppliedByWalls = 500;
+        if(collision.gameObject.tag.Equals("Walls"))
+        {
+            Vector3 currentVelocity = sphere.GetComponent<Rigidbody>().velocity;
+            Vector3 reflectedVector = Vector3.Reflect(collision.contacts[0].normal, currentVelocity);
```

Comment:
We can't use the currentVelocity quite like this.  We have to save the "oldVelocity" in FixedUpdate() and then pass in oldVelocity instead of currentVelocity.  Don't ask me why. . . Unity magic.  >.>

http://answers.unity3d.com/questions/352609/how-can-i-reflect-a-projectile.html